### PR TITLE
[Layout foundations] Deprecate TextContainer

### LINKED ISSUE #7404: [Layout foundations] Deprecate `TextContainer` and replace usage with new layout primitives

### PR DESCRIPTION
**Pull request by kyledurand**
Closes: https://github.com/Shopify/polaris/issues/7404

### CODE REVIEW

File: polaris-react/src/components/CalloutCard/CalloutCard.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import {CancelSmallMinor} from '@shopify/polaris-icons';
 import {classNames} from '../../utilities/css';
 import type {Action} from '../../types';
 import {Card} from '../Card';
+// eslint-disable-next-line import/no-deprecated
 import {TextContainer} from '../TextContainer';
```
Comment (aveline) [inline polaris-react/src/components/CalloutCard/CalloutCard.tsx:8]: Should this get replaced by `AlphaStack`? Or will that happen in another PR?

**Review by aveline — APPROVED** [review]

File: polaris-react/src/components/CalloutCard/CalloutCard.tsx (hunk 1 above)
Comment (kyledurand) [reply polaris-react/src/components/CalloutCard/CalloutCard.tsx:8]: That will happen in the migration PR which @yurm04 is working on

**Review by laurkim — APPROVED** [review]
💯 
Probably need to update URL for redirects

File: polaris.shopify.com/content/components/deprecated/text-container.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ keywords:
   - vertical
   - margin
   - padding
+status:
+  value: Deprecated
```
Comment (laurkim) [inline polaris.shopify.com/content/components/deprecated/text-container.md:15]: category should also be updated to deprecated

**Pull request merged**